integration/src/test/resources/templates/kmp-with-toolchain/build.gradle

```diff
@@ -0,0 +1,25 @@
+kotlin {
+    jvm {
+        compilations.create('benchmark') { associateWith(compilations.main) }
+        jvmToolchain(21)
+    }
+
+    sourceSets {
+        commonMain {
+            dependencies {
+                implementation("org.jetbrains.kotlinx:kotlinx-benchmark-runtime:0.5.0-SNAPSHOT")
+            }
+        }
+        jvmMain {
+        }
+        jvmBenchmark {
+            dependsOn(jvmMain)
+        }
+    }
+}
+
+benchmark {
+    targets {
+        register("jvm")
```

Comment:
> It would be nice if a guide was provided on how to specify toolchains for JVM benchmarks

This particular task does not facilitate per-configuration toolchains setup, but instead forces all the kotlinx-benchmark's tasks dealing with JVM/JMH to use the toolchain already configured for a project (i.e. it's about the original problem from https://github.com/Kotlin/kotlinx-benchmark/issues/176, not about the wish to setup multiple configs running benchmarks with different JVMs).

Or do you mean that it's worth documenting how to configure toolchains in general?

> Is the decision to create a benchmark compilation and yet register the main compilation for benchmarking deliberate?

Not really, I'll clean it up.

Reply:
I misunderstood the use of `org.gradle.toolchains.foojay-resolver-convention` plugin. I thought its is a part of solution to align Java executables between project and benchmark launcher.